SU2_CFD/src/variables/CTurbVariable.cpp

```diff
@@ -35,8 +35,61 @@ CTurbVariable::CTurbVariable(unsigned long npoint, unsigned long ndim, unsigned
     turb_index.resize(nPoint) = su2double(1.0);
     intermittency.resize(nPoint) = su2double(1.0);
 
+    DES_LengthScale.resize(nPoint) = su2double(0.0);
+    SRSGridSize.resize(nPoint) = su2double(0.0);
+    
+    Vortex_Tilting.resize(nPoint);
+
    }
 
+
+void CTurbVariable::SetVortex_Tilting(unsigned long iPoint, CMatrixView<const su2double> PrimGrad_Flow,
+                                        const su2double* Vorticity, su2double LaminarViscosity) {
+
+  su2double Strain[3][3] = {{0,0,0}, {0,0,0}, {0,0,0}}, Omega, StrainDotVort[3], numVecVort[3];
+  su2double numerator, trace0, trace1, denominator;
+
+  AD::StartPreacc();
+  AD::SetPreaccIn(PrimGrad_Flow, nDim+1, nDim);
+  AD::SetPreaccIn(Vorticity, 3);
+  /*--- Eddy viscosity ---*/
+  AD::SetPreaccIn(muT(iPoint));
+  /*--- Laminar viscosity --- */
+  AD::SetPreaccIn(LaminarViscosity);
+
+  Strain[0][0] = PrimGrad_Flow[1][0];
```

Comment:
We need to abstract the index for velocity somehow, otherwise this is not compatible with the NEMO solver.
Maybe change PrimGrad_Flow to VelGrad and whoever call this function is responsible by passing the correct part of the gradients.